src/axom/inlet/inlet_utils.cpp

```diff
@@ -165,5 +189,34 @@ void markAsStructCollection(axom::sidre::Group& target)
   }
 }
 
+void markRetrievalStatus(axom::sidre::Group& target, const ReaderResult result)
+{
+  if(result != ReaderResult::Success)
+  {
+    target.createViewScalar("retrieval_status", static_cast<int>(result));
+  }
+}
+
+ReaderResult collectionRetrievalResult(const bool homogeneous,
+                                       const bool none_of_requested_type)
+{
+  // First check if the collection was entirely the wrong type
+  if(!homogeneous && none_of_requested_type)

```

Comment:
Reading this comment... should this then be "homogeneous && none_of_requested_type" or is the comment wrong?

Reply:
homogeneous is more like "none_of_other_type", I'll rename it